sccp/sccp-impl/src/main/java/org/mobicents/protocols/ss7/sccp/impl/router/RuleImpl.java

```diff
@@ -472,6 +496,31 @@ public boolean matches(SccpAddress address, boolean isMtpOriginated, int msgNetw
         }
     }
 
+    /**
+     * Checks if SSN matches between rule address pattern and provided address
+     * @param address - a provided address to match
+     * @param pattern - a rule pattern address
+     * @return true if SSN is present in both patern and received addresses and they are the same
+     *              or pattern has SSN flag unset in AI (bit 7)(isSsnPresent = false for pattern)
+     */
+    private boolean isSsnMatch(SccpAddress address, SccpAddress pattern) {
+        if(pattern.getAddressIndicator().isSSNPresent() && address.getAddressIndicator().isSSNPresent()) {
+           if(address.getSubsystemNumber() == pattern.getSubsystemNumber()) {
+               return true;
+           }
+        } else if(!pattern.getAddressIndicator().isSSNPresent()) {
```

Comment:
@vetss This condition causes the matching algorithm matches any destination address SSN in case pattern AI SSNpresent bit is not set.